bundles/org.dataflowanalysis.analysis.dfd/src/org/dataflowanalysis/analysis/dfd/core/DFDTransposeFlowGraphFinder.java

```diff
@@ -81,13 +85,49 @@ public List<? extends AbstractTransposeFlowGraph> findTransposeFlowGraphs(List<?
      * @param inputPins Relevant input pins on the given vertex
      * @return List of sinks created from the initial sink with previous vertices calculated
      */
-    private List<DFDVertex> determineSinks(DFDVertex sink, List<Pin> inputPins, List<Node> sourceNodes) {
+    private List<DFDVertex> determineSinks(DFDVertex sink, List<Pin> pins, List<Node> sourceNodes) {
         List<DFDVertex> vertices = new ArrayList<>();
         vertices.add(sink);
 
         if (sourceNodes.contains(sink.getReferencedElement())) {
             return vertices;
         }
+        
+        var inputPins = new ArrayList<>(pins);
+        
+        Map<Pin, List<Pin>> inToPreviousNodeInPinsMap = new HashMap<>();
+        inputPins.forEach(pin -> {
+        	Set<Pin> outputPins = new HashSet<>();
+        	inToPreviousNodeInPinsMap.put(pin, new ArrayList<>());
+        	dataFlowDiagram.getFlows()
+            .stream()
+            .filter(flow -> flow.getDestinationPin()
+                    .equals(pin))
+            .forEach(flow -> {
+            	outputPins.add(flow.getSourcePin());
+            });
+        	outputPins.stream().forEach(outPin -> {
+        		Behaviour behaviour = (Behaviour) outPin.eContainer();
+        		behaviour.getAssignment().stream().filter(it -> it.getOutputPin().equals(outPin)).forEach(it -> inToPreviousNodeInPinsMap.get(pin).addAll(it.getInputPins()));
+        	});
+        });
```

Comment:
I found this kinda difficult to read due to awkward line breaks, 
maybe just change to standard for each loops to avoid this?

Reply:
Changed